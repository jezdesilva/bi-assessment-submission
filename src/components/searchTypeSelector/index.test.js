import React from 'react';
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { SearchTypeSelector } from './index';
import {
  SEARCH_TITLE,
  SEARCH_AUTHOR,
} from '../../constants';

describe('SearchTypeSelector', () => {
  let history;

  beforeEach(() => {
    history = createMemoryHistory();
  });

  test('renders without crashing', () => {
    const { container } = render(
      <Router history={history}>
        <SearchTypeSelector searchType={SEARCH_TITLE} />
      </Router>
    );
    expect(container.innerHTML).toMatch('searchTypeSelector');

  });

  test('should match the snapshot', () => {
    const { container } = render(
      <Router history={history}>
        <SearchTypeSelector searchType={SEARCH_TITLE} />
      </Router>
    );
    expect(container.innerHTML).toMatchSnapshot();
  });

  test('highlights the default selected type', () => {
    const { container } = render(
      <Router history={history}>
        <SearchTypeSelector />
      </Router>
    );

    const titleButton = container.querySelector('button[name="title"]');
    expect(titleButton.outerHTML).toMatch('active');
  });

  test('highlights the currently selected type', () => {
    const { container } = render(
      <Router history={history}>
        <SearchTypeSelector searchType={SEARCH_AUTHOR} />
      </Router>
    );

    const titleButton = container.querySelector('button[name="author"]');
    expect(titleButton.outerHTML).toMatch('active');
  });

  test('handles a cick on search type button correctly', () => {
    const { getByText } = render(
      <Router history={history}>
        <SearchTypeSelector searchType={SEARCH_TITLE} />
      </Router>
    );

    fireEvent.click(getByText(/Author/i));
    expect(history.entries.pop().pathname).toMatch('author');
  })
});