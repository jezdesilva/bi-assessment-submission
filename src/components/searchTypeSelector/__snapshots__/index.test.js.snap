// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTypeSelector should match the snapshot 1`] = `"<div class=\\"ui right floated buttons searchTypeSelector\\"><button name=\\"title\\" class=\\"ui active button\\"><i aria-hidden=\\"true\\" class=\\"book icon\\"></i>Title</button><div class=\\"or\\"></div><button name=\\"author\\" class=\\"ui button\\"><i aria-hidden=\\"true\\" class=\\"user icon\\"></i>Author</button><div class=\\"or\\"></div><button name=\\"q\\" class=\\"ui button\\"><i aria-hidden=\\"true\\" class=\\"check icon\\"></i>Everything</button></div>"`;
